tests/cross_functional/conftest.py

```
@@ -2965,3 +2973,409 @@ def _factory(
+                f"Failed to reach target cluster capacity - alerts {expected_alerts} not triggered"
+            )
+
+        # Count pending clones if minimum is specified
+        if min_pending_clones > 0:
+            pending_clones = 0
+            for clone in all_clones[-15:]:  # Check last 15 clones
+                try:
+                    if clone.status != constants.STATUS_BOUND:
+                        pending_clones += 1
+                except Exception:
+                    pending_clones += 1
```

Ruff (0.15.15)

[warning] `try`-`except` within a loop incurs performance overhead

(PERF203)